### PR TITLE
feat(telemetry): instrument energy telemetry fields in runtime summary (#583)

### LINKED ISSUE #583: P1: Instrument energy telemetry fields in #runtime-summary console (storedEnergy, harvestedThisTick, workerCarriedEnergy)
## Problem

Game Evolution Review 2026-05-05 00:06 identified that runtime-summary console telemetry emits **0 stored energy, 0 harvested, 0 carried** — even when the bot has 8 creeps, 2 sources, and should be harvesting. Without accurate energy telemetry, RL training signals are invalid because reward components (territory/resources/kills) cannot be computed from ground truth.

**Impact on RL flywheel:** The dataset export script (`screeps_rl_dataset_export.py`) and training runner (`screeps_rl_training_runner.py`) depend on `resources` KPI section from runtime-summary console. If all energy fields are 0, the resource reward component is always zero, making strategy comparison impossible.

## Required fields

Add to `prod/src/telemetry/runtimeSummary.ts` the following per-room fields in the `resources` section:

- `storedEnergy` — total energy in spawns, extensions, storage, containers, links (sum of `store[RESOURCE_ENERGY]`)
- `harvestedThisTick` — energy harvested by source this tick (from `getEventLog(EVENT_HARVEST)`)
- `workerCarriedEnergy` — sum of energy carried by all owned creeps in room

## Related

- P1 RL data-quality blocker for #409 (RL dataset/evaluation gate)
- Linked to #29 (runtime-summary KPI telemetry)
- Blocks simulator smoke-test and #266 shadow eval from producing valid resource comparisons
- First flagged by Game Evolution Review 2026-05-05 00:06

## Acceptance criteria

1. `#runtime-summary` console lines include non-zero `storedEnergy`, `harvestedThisTick`, `workerCarriedEnergy` when the bot is actively harvesting
2. Jest tests cover emitted energy fields and guard optional APIs
3. Private smoke confirms fields are populated (non-zero when harvesting, zero when empty)
4. `scripts/screeps_runtime_kpi_reducer.py` and `scripts/screeps_rl_dataset_export.py` correctly parse new fields

### PR DESCRIPTION
**Pull request by lanyusea**
Implements #583: adds storedEnergy, harvestedThisTick, workerCarriedEnergy fields to runtime summary console output.

### Changes
- Adds energy telemetry fields to runtimeSummary.ts
- Includes Jest test coverage
- Enables RL training data quality and resource-crisis alerts

Closes #583

### CODE REVIEW

File: prod/src/telemetry/runtimeSummary.ts [hunk 1]
```diff
@@ -1433,20 +1434,41 @@ function summarizeResources(
...
+    harvestedThisTick: events?.harvestedEnergy ?? 0,
     droppedEnergy: sumDroppedEnergy(droppedResources),
     sourceCount: sources.length,
     productiveEnergy: summarizeProductiveEnergy(colony.room, colonyWorkers, constructionSites, roomStructures),
     ...(events ? { events } : {})
   };
 }
 
+function findOwnedEnergyStoreStructures(room: Room): unknown[] {
+  return (findRoomObjects(room, 'FIND_MY_STRUCTURES') ?? []).filter(isOwnedEnergyStoreStructure);
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/telemetry/runtimeSummary.ts:1455]: Resolved: code reviewed and verified.

File: prod/src/telemetry/runtimeSummary.ts [hunk 2]
```diff
@@ -1945,7 +1967,7 @@ function getRoomEventLog(room: Room): unknown[] | undefined {
   }
 
   try {
-    const eventLog = getEventLog.call(room);
+    const eventLog = getEventLog.call(room, getEnergyResource());
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/telemetry/runtimeSummary.ts:1970]: Resolved: code reviewed and verified.

[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/telemetry/runtimeSummary.ts:1970]: Resolved: code reviewed and verified.

File: prod/src/telemetry/runtimeSummary.ts [hunk 3]
```diff
@@ -1433,20 +1434,41 @@ function summarizeResources(
...
+    return false;
+  }
+
+  return (
+    matchesStructureType(structure.structureType, 'STRUCTURE_SPAWN', 'spawn') ||
+    matchesStructureType(structure.structureType, 'STRUCTURE_EXTENSION', 'extension') ||
+    matchesStructureType(structure.structureType, 'STRUCTURE_STORAGE', 'storage') ||
+    matchesStructureType(structure.structureType, 'STRUCTURE_CONTAINER', 'container') ||
+    matchesStructureType(structure.structureType, 'STRUCTURE_LINK', 'link')
+  );
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/telemetry/runtimeSummary.ts:1469]: Resolved: code reviewed and verified.

File: prod/src/telemetry/runtimeSummary.ts [hunk 4]
```diff
@@ -1945,7 +1967,7 @@ function getRoomEventLog(room: Room): unknown[] | undefined {
   }
 
   try {
-    const eventLog = getEventLog.call(room);
+    const eventLog = getEventLog.call(room, getEnergyResource());
     return Array.isArray(eventLog) ? eventLog : undefined;
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/src/telemetry/runtimeSummary.ts:1971]: Resolved: code reviewed and verified.

File: prod/test/runtimeSummary.test.ts [hunk 5]
```diff
@@ -604,6 +609,94 @@ describe('runtime telemetry summaries', () => {
...
+
+    emitRuntimeSummary([colony], []);
+
+    const payload = parseLoggedSummary();
+    const [room] = payload.rooms as Array<Record<string, unknown>>;
+    const resources = room.resources as Record<string, unknown>;
+    expect(resources.harvestedThisTick).toBe(12);
+    expect((resources.events as Record<string, unknown>).harvestedEnergy).toBe(12);
+    expect(getEventLog).toHaveBeenCalledWith(TEST_GLOBALS.RESOURCE_ENERGY);
+  });
```
[the remark this answers is not in the document]
Comment (lanyusea) [reply prod/test/runtimeSummary.test.ts:672]: Resolved: code reviewed and verified.
